### PR TITLE
[xdl][expo-cli] add the "target" warning for export

### PR DESCRIPTION
**Pull request by cruzach**
## Why

we warn if you run `expo publish` in a bare project without the `--target` flag. So we should do the same for `expo export` 

to keep things in one place, i moved the `logBareWorkflowWarnings` code from `expo-cli/publish.ts` to `xdl/Project.ts`, since that's where we do the relevant publishing/exporting work. This also prevents us from calling `getConfig` (which is expensive) more times than we need to

Tested in a bare project:
- `expo export` logs warning ✅ 
- `expo publish` logs warning ✅ 
- `expo export --target bare` does not log warning ✅ 
- `expo publish --target bare` does not log warning ✅ 
- remove `expo` from dependencies in package.json, does not log warning ✅ 

Tested in a managed project:
- never logs warning✅

### CODE REVIEW

File: packages/xdl/src/Project.ts [hunk 1]
```diff
@@ -2049,3 +2058,32 @@ export async function stopAsync(projectDir: string): Promise<void> {
...
+ * it will run in Expo client in development even. We should revisit this with
+ * dev client, and possibly also by excluding SDK version for bare
+ * expo-updates usage in the future (and then surfacing this as an error in
+ * the Expo client app instead)
+ *
+ * Related: https://github.com/expo/expo/issues/9517
+ *
+ * @param pkg package.json
+ */
+export function logBareWorkflowWarnings(pkg: PackageJSONConfig) {
```
Comment (EvanBacon) [inline packages/xdl/src/Project.ts:2075]: no new code in Project.ts

**Review by EvanBacon — CHANGES_REQUESTED** [review]
see comment

File: packages/xdl/src/Project.ts (hunk 1 above)
Comment (cruzach) [reply packages/xdl/src/Project.ts]: ah okay, didn't know that 👍

**Comment by EvanBacon** [comment]
Is this still relevant now that selecting workflows are deprecated (at least for builds).

**Pull request closed without merging**